fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckContent.kt

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.ui
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.BottomSheetHandle
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private val bottomSheetShape = RoundedCornerShape(topStart = 16.dp, topEnd = 16.dp)
+private const val BOTTOM_SHEET_HANDLE_WIDTH_PERCENT = 0.1f
+
+/**
+ * A composable to display content for ReviewQualityCheck bottom sheet.
+ *
+ * @param onRequestDismiss Invoked when a user actions requests dismissal of the bottom sheet.
+ */
+@Composable
+fun ReviewQualityCheckContent(
+    onRequestDismiss: () -> Unit,
+) {
+    Column(
+        modifier = Modifier
+            .background(
+                color = FirefoxTheme.colors.layer1,
+                shape = bottomSheetShape,
+            )
+            .padding(16.dp),
+    ) {
+        BottomSheetHandle(
+            onRequestDismiss = onRequestDismiss,
+            contentDescription = stringResource(R.string.browser_menu_review_quality_check_close),
+            modifier = Modifier
+                .fillMaxWidth(BOTTOM_SHEET_HANDLE_WIDTH_PERCENT)
+                .align(Alignment.CenterHorizontally),
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        Header()
+
+        Spacer(modifier = Modifier.height(16.dp))
+    }
+}
+
+@Composable
+private fun Header() {
```

Comment:
I'm on the fence about having a separate Composable function for something that isn't reused and doesn't have any business logic, but this is also something we can circle back to later, as this UI develops.

Reply:
Just adding my 2 cents - Composable's can become somewhat monolithic, I think it can be useful to break it up into logical components like this as it can make it easier to digest

Reply:
Adding to what Tom said, I agree it becomes harder to read and breaking it down in more contained components makes it more manageable. Also, the header seems to have different states as the bottom sheet is expanded or partial, so it'll be a nice abstraction there with some animation perhaps. Detekt also flags long functions so that fed into this decision as well, I think it'll become more clearer as the UI develops and we can see what makes more sense then.